nucypher/blockchain/eth/registry.py

```diff
@@ -231,33 +188,93 @@ def search(self, contract_name: str=None, contract_address: str=None):
         return contracts if contract_name else contracts[0]
 
 
-class LocalEthereumContractRegistry(EthereumContractRegistry):
+class LocalContractRegistry(BaseContractRegistry):
+
+    def __init__(self, filepath: str, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.__filepath = filepath
+        self.log.info("Using contract registry {}".format(self.__filepath))
+
+    def __repr__(self):
+        r = f"{self.__class__.__name__}(filepath={self.filepath})"
+        return r
+
+    @property
+    def filepath(self) -> str:
+        return self.__filepath
+
+    def _swap_registry(self, filepath: str) -> bool:
+        self.__filepath = filepath
+        return True
+
+    def read(self) -> Union[list, dict]:
+        """
+        Reads the registry file and parses the JSON and returns a list.
+        If the file is empty it will return an empty list.
+        If you are modifying or updating the registry file, you _must_ call
+        this function first to get the current state to append to the dict or
+        modify it because _write_registry_file overwrites the file.
+        """
+        try:
+            with open(self.filepath, 'r') as registry_file:
+                self.log.debug("Reading from registrar: filepath {}".format(self.filepath))
+                registry_file.seek(0)
+                file_data = registry_file.read()
+                if file_data:
+                    try:
+                        registry_data = json.loads(file_data)
+                    except JSONDecodeError:
+                        raise self.RegistryError(f"Registry contains invalid JSON at '{self.__filepath}'")
+                else:
+                    registry_data = list() if self._multi_contract else dict()
+
+        except FileNotFoundError:
+            raise self.NoRegistry("No registry at filepath: {}".format(self.filepath))
+
+        except JSONDecodeError:
+            raise
+
+        return registry_data
+
+    def write(self, registry_data: list) -> None:
+        """
+        Writes the registry data list as JSON to the registry file. If no
+        file exists, it will create it and write the data. If a file does exist
+        it will _overwrite_ everything in it.
+        """
+        # Ensure parent path exists
+        os.makedirs(abspath(dirname(self.__filepath)), exist_ok=True)
 
-    _default_registry_filepath = os.path.join(
-        DEFAULT_CONFIG_ROOT, 'dev_contract_registry.json'
-    )
+        with open(self.__filepath, 'w+') as registry_file:
```

Comment:
Since we remove the old data anyway, why not simply:
```
with open(self.__filepath, 'w') as f:
    json.dump(registry_data, f)
```

Reply:
:heavy_check_mark: